conans/client/remote_manager.py

```diff
@@ -328,6 +357,10 @@ def _call_remote(self, remote, method, *argc, **argv):
         except ConnectionError as exc:
             raise ConanConnectionError("%s\n\nUnable to connect to %s=%s"
                                        % (str(exc), remote.name, remote.url))
+        except RecipeNotFoundException as exc:
+            raise NotFoundException("%s. [Remote: %s]" % (exception_message_safe(exc), remote.name))
```

Comment:
What? So if calling a remote returns a RecipeNotFound you re-convert it to NotFoundException? Seems very weird.

Reply:
The `RecipeNotFoundException` accepts a `ref`, not a String, so if we want to show the `[Remote:` it has to be converted or support the remote also in the `RecipeNotFoundException` or in the `NotFoundException`. Let me know what do you prefer.

Reply:
I think I prefer not converting back to a more generic exception. So I'd say to either:
- Add a "remote" attribute to the RecipeNotFound
- Make RecipeNotFound have an internal string attribute, remove "ref" attribute. The string attribute is still created at construction time from "ref" parameter.

In any case, capture RecipeNotFound, re-launch as RecipeNotFound